pkg/rancher-desktop/pages/extensions/installed.vue

```diff
@@ -66,35 +64,21 @@ export default Vue.extend({
     extensionTitle(ext: {id: string, labels: Record<string, string>}): string {
       return ext.labels?.['org.opencontainers.image.title'] ?? ext.id;
     },
-    async uninstall(installed: ExtensionWithId) {
+    async uninstall(installed: ExtensionState) {
       this.$set(this.busy, installed.id, true);
       try {
         await this.$store.dispatch('extensions/uninstall', { id: installed.id });
       } finally {
         this.$delete(this.busy, installed.id);
       }
     },
-    canUpgrade(installed: ExtensionWithId) {
-      const available = this.marketData.find(item => item.slug === installed.id);
-
-      try {
-        return available && semver.gt(available.version, installed.version, { loose: true });
-      } catch (ex) {
-        // If there is invalid semver, don't allow upgrades.
-        return false;
-      }
-    },
-    async upgrade(installed: ExtensionWithId) {
-      const available = this.marketData.find(item => item.slug === installed.id);
-
-      if (!available) {
-        console.error(`Failed to upgrade ${ installed.id }: no version found in catalog`);
+    async upgrade(installed: ExtensionState) {
+      const id = `${ installed.id }:${ installed.availableVersion }`;
 
+      if (!installed.availableVersion) {
+        // Should not have reached here.
```

Comment:
Do we need a log line so we know what was the cause of the return?

Reply:
I don't think so; if `availableVersion` isn't set, then `upgrade` shouldn't have been called in the first place (because that button is only rendered if `row.canUpgrade` which is only set if `availableVersion` is set)?